test/modules/vpc.go

```diff
@@ -0,0 +1,262 @@
+package modules
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	aws_sdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/assert"
+)
+
+// DeployVpcUsingTerraform deploys the Terraform code in the given working dir and returns the Terraform output
+func DeployVpcUsingTerraform(t *testing.T, workingDir string) {
+	// Get a random AWS region
+	awsRegion := aws.GetRandomStableRegion(t, []string{"us-east-1", "us-east-2"}, nil)
+	test_structure.SaveString(t, workingDir, "awsRegion", awsRegion)
+
+	// Construct the terraform options with default retryable errors to handle the most common retryable errors in
+	// terraform testing.
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: workingDir,
+		Vars: map[string]interface{}{
+			"region": awsRegion,
+		},
+	})
+
+	// Save the options so later test stages can use them
+	test_structure.SaveTerraformOptions(t, workingDir, terraformOptions)
+}
+
+func ValidateOnlyPublicSubnets(t *testing.T, workingDir string) {
+	// Load the terraform options
+	terraformOptions := test_structure.LoadTerraformOptions(t, workingDir)
+	awsRegion := test_structure.LoadString(t, workingDir, "awsRegion")
+
+	// Create aws session
+	session, err := session.NewSession()
+	assert.NoError(t, err, "Error creating AWS session")
+
+	// Create Client
+	ec2Client := ec2.New(session, &aws_sdk.Config{Region: aws_sdk.String(awsRegion)})
+
+	// Check that the VPC exists
+	vpcID := assertVpcExists(t, terraformOptions, awsRegion)
+
+	// Assert that the VPC has the correct number of subnets
+	numAzs, err := strconv.Atoi(terraform.Output(t, terraformOptions, "num_availability_zones"))
+	assert.NoError(t, err, "Error converting num_availability_zones to int")
+	assertVpcHasCorrectNumberOfSubnets(t, terraformOptions, awsRegion, "public_subnet_ids", numAzs)
+
+	// Assert that the VPC has no private subnets
+	subnets, err := ec2Client.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws_sdk.String("vpc-id"),
+				Values: []*string{aws_sdk.String(vpcID)},
+			},
+			{
+				Name:   aws_sdk.String("tag:Name"),
+				Values: []*string{aws_sdk.String("Private Subnet")},
```

Comment:
The Name tag for a private subnet is formatted like `"${var.vpc_name}-private-%s"` likewise, the public subnet is formatted like `"${var.vpc_name}-public-%s"` where the `%s` is the availability zone name.